simplified-books-borrowing/src/main/java/org/nypl/simplified/books/borrowing/internal/BorrowHTTP.kt

```diff
@@ -0,0 +1,216 @@
+package org.nypl.simplified.books.borrowing.internal
+
+import one.irradia.mime.api.MIMECompatibility
+import one.irradia.mime.api.MIMEType
+import org.librarysimplified.http.api.LSHTTPAuthorizationBasic
+import org.librarysimplified.http.api.LSHTTPAuthorizationBearerToken
+import org.librarysimplified.http.api.LSHTTPAuthorizationType
+import org.librarysimplified.http.api.LSHTTPProblemReport
+import org.librarysimplified.http.downloads.LSHTTPDownloadRequest
+import org.librarysimplified.http.downloads.LSHTTPDownloadState
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.DownloadReceiving
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.DownloadStarted
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.LSHTTPDownloadResult.DownloadCancelled
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.LSHTTPDownloadResult.DownloadCompletedSuccessfully
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.LSHTTPDownloadResult.DownloadFailed.DownloadFailedExceptionally
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.LSHTTPDownloadResult.DownloadFailed.DownloadFailedServer
+import org.librarysimplified.http.downloads.LSHTTPDownloadState.LSHTTPDownloadResult.DownloadFailed.DownloadFailedUnacceptableMIME
+import org.nypl.simplified.accounts.api.AccountAuthenticationCredentials
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountLoggedIn
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountLoggingIn
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountLoggingInWaitingForExternalAuthentication
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountLoggingOut
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountLoginFailed
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountLogoutFailed
+import org.nypl.simplified.accounts.api.AccountLoginState.AccountNotLoggedIn
+import org.nypl.simplified.accounts.api.AccountReadableType
+import org.nypl.simplified.books.borrowing.BorrowContextType
+import org.nypl.simplified.books.borrowing.subtasks.BorrowSubtaskException.BorrowSubtaskFailed
+import java.io.File
+import java.net.URI
+
+/**
+ * Convenience functions over HTTP.
+ */
+
+object BorrowHTTP {
+
+  /**
+   * Encode the given problem report as a set of presentable attributes.
+   */
+
+  fun problemReportAsAttributes(
+    problemReport: LSHTTPProblemReport?
+  ): Map<String, String> {
+    return when (problemReport) {
+      null -> mapOf()
+      else -> {
+        val attributes = mutableMapOf<String, String>()
```

Comment:
You could make this a bit more fluent using the `to` infix operator:

```
else -> mapOf(
  "HTTP problem detail" to problemReport.detail ?: ""
  "HTTP problem status" to problemReport.status.toString()
  // ...
)
```

Little bit of syntactic sugar but otherwise this looks fine as-is.